app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt

```diff
@@ -1,52 +1,129 @@
 package com.example.mykotlincalculator21.mvp.model
 
 import com.example.mykotlincalculator21.mvp.contract.CalculatorContract
-import com.example.mykotlincalculator21.utils.EMPTY_STRING
-import com.example.mykotlincalculator21.utils.ERROR_MESSAGE
-import com.example.mykotlincalculator21.utils.OPERATOR_DIVIDE
-import com.example.mykotlincalculator21.utils.OPERATOR_MINUS
-import com.example.mykotlincalculator21.utils.OPERATOR_MULTIPLY
-import com.example.mykotlincalculator21.utils.OPERATOR_SUM
+import com.example.mykotlincalculator21.utils.ResultUtils
+import com.example.mykotlincalculator21.utils.NumbersUtils.POSITION_ONE
+import com.example.mykotlincalculator21.utils.NumbersUtils.POSITION_ZERO
+import com.example.mykotlincalculator21.utils.NumbersUtils.ZERO_NUMBER_DOUBLE_TYPE
+import com.example.mykotlincalculator21.operand.Operand
+import com.example.mykotlincalculator21.utils.StringUtils.EMPTY_STRING
+import com.example.mykotlincalculator21.utils.StringUtils.OPERATOR_DIVIDE
+import com.example.mykotlincalculator21.utils.StringUtils.OPERATOR_MINUS
+import com.example.mykotlincalculator21.utils.StringUtils.OPERATOR_MULTIPLY
+import com.example.mykotlincalculator21.utils.StringUtils.OPERATOR_SUM
 
 class CalculatorModel : CalculatorContract.CalculatorModelContract {
-    private var firstOperand: String = EMPTY_STRING
-    private var secondOperand: String = EMPTY_STRING
+    private val firstOperandUtils = Operand()
+    private val secondOperandUtils = Operand()
     private var operator: String = EMPTY_STRING
-    private var result: String = EMPTY_STRING
+    private var resultOperation: String = EMPTY_STRING
+    private var result = ResultUtils.SUCCESS
 
     override fun saveNumber(number: String) {
         if (operator.isEmpty()) {
-            firstOperand += number
+            firstOperandUtils.addNumber(number)
         } else {
-            secondOperand += number
+            secondOperandUtils.addNumber(number)
         }
     }
 
-    override fun getPartialResult() = "$firstOperand$operator$secondOperand"
+    override fun getValue(): String {
+        var firstValue: String = (firstOperandUtils.getValue()).toString()
+        var secondValue: String = (secondOperandUtils.getValue()).toString()
+        if (secondOperandUtils.isEmpty()) {
+            secondValue = EMPTY_STRING
+        }
+        return "$firstValue$operator$secondValue"
+    }
+
+    private fun isValidOperation(): Boolean = when {
+        operator.isEmpty() && firstOperandUtils.isEmpty() -> {
+            resultOperation = EMPTY_STRING
```

Comment:
why this is a boolean funtion called isValidOperation and it sets also de Result?

when you call this function (line 61) you are setting again the result and the error. i think you can remove the writing of result here. right?

Reply:
Done, thx